karapace/kafka/producer.py

```diff
@@ -59,3 +62,41 @@ def partitions_for(self, topic: str) -> dict[int, PartitionMetadata]:
             return self.list_topics(topic).topics[topic].partitions
         except KafkaException as exc:
             raise_from_kafkaexception(exc)
+
+
+class AIOKafkaProducer:
+    """An async wrapper around `KafkaProducer` built on top of confluent-kafka.
+
+    Starting `AIOKafkaProducer` instantiates a `KafkaProducer` and starts a poll-thread.
+
+    The poll-thread continuously polls the underlying producer so buffered messages
+    are sent and asyncio futures returned by the `send` method can be awaited.
+    """
+
+    def __init__(
+        self,
+        bootstrap_servers: Iterable[str] | str,
```

Comment:
thought (non-blocking): Just noting that `str` is a subtype of `Iterable[str]`, so technically we could safely replace with just `Iterable[str]`, but let's not do that ;)

On a tangent, if we wanted to be super stringent here, we could bring in this construct to model "sequence, but not a `str`": https://github.com/hauntsaninja/useful_types/blob/951d36b2aaaee6ab9fa1efcb8c4f1e2733247ff5/useful_types/__init__.py#L285.